sentry-android-core/src/main/java/io/sentry/android/core/AndroidCpuCollector.java

```diff
@@ -0,0 +1,115 @@
+// The content of the /proc/self/stat file is specified in
+// https://man7.org/linux/man-pages/man5/proc.5.html
+@ApiStatus.Internal
+public final class AndroidCpuCollector implements ICpuCollector {
+
+  private long lastRealtimeNanos = 0;
+  private long lastCpuNanos = 0;
+
+  /** Number of clock ticks per second. */
+  private long clockSpeedHz = 1;
+
+  private long numCores = 1;
+  private final long NANOSECOND_PER_SECOND = 1_000_000_000;
+
+  /** Number of nanoseconds per clock tick. */
+  private double nanosecondsPerClockTick = NANOSECOND_PER_SECOND / (double) clockSpeedHz;
+
+  /** File containing stats about this process. */
+  private final @NotNull File selfStat = new File("/proc/self/stat");
+
+  private final @NotNull ILogger logger;
+  private final @NotNull BuildInfoProvider buildInfoProvider;
+  private boolean isEnabled = false;
+
+  public AndroidCpuCollector(
+      final @NotNull ILogger logger, final @NotNull BuildInfoProvider buildInfoProvider) {
+    this.logger = Objects.requireNonNull(logger, "Logger is required.");
+    this.buildInfoProvider =
+        Objects.requireNonNull(buildInfoProvider, "BuildInfoProvider is required.");
+  }
+
+  @SuppressLint("NewApi")
+  @Override
+  public void setup() {
+    if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.LOLLIPOP) {
+      isEnabled = false;
+      return;
+    }
+    isEnabled = true;
+    clockSpeedHz = Os.sysconf(OsConstants._SC_CLK_TCK);
+    numCores = Os.sysconf(OsConstants._SC_NPROCESSORS_CONF);
+    nanosecondsPerClockTick = NANOSECOND_PER_SECOND / (double) clockSpeedHz;
+    lastCpuNanos = readTotalCpuNanos();
+  }
+
+  @SuppressLint("NewApi")
+  @Override
+  public @Nullable CpuCollectionData collect() {
+    if (buildInfoProvider.getSdkInfoVersion() < Build.VERSION_CODES.LOLLIPOP || !isEnabled) {
+      return null;
+    }
+    long nowNanos = SystemClock.elapsedRealtimeNanos();
+    long realTimeNanosDiff = nowNanos - lastRealtimeNanos;
+    lastRealtimeNanos = nowNanos;
+    long cpuNanos = readTotalCpuNanos();
+    long cpuNanosDiff = cpuNanos - lastCpuNanos;
+    lastCpuNanos = cpuNanos;
+    // Later we need to divide the percentage by the number of cores, otherwise we could
+    // get a percentage value higher than 1. We also want to send the percentage as a
+    // number from 0 to 100, so we are going to multiply it by 100
+    double cpuUsagePercentage = cpuNanosDiff / (double) realTimeNanosDiff;
+
+    return new CpuCollectionData(
+        System.currentTimeMillis(), (cpuUsagePercentage / (double) numCores) * 100.0);
+  }
+
+  /** Read the /proc/self/stat file and parses the result. */
+  private long readTotalCpuNanos() {
+    String stat = null;
+    try {
+      stat = FileUtils.readText(selfStat);
+    } catch (IOException e) {
+      // If an error occurs when reading the file, we avoid reading it again until the setup method
+      // is called again
+      isEnabled = false;
+      logger.log(
```

Comment:
`m`: Should we change the log level to `warning` perhaps? It's not some critical functionality of the SDK, so probably fine to only log it when `debug` is enabled